### DISCUSSION TITLE
FFMPEG Tuning - Best Practice

### DISCUSSION THREAD

**Discussion by AdiAbuAli** [mrlt8/docker-wyze-bridge#193] [Show and tell]
Hi,
I recently had issue with one V3 with bad Wi-Fi coverage and a lot of lost frames, caused FFMPEG to freeze and broke all other stream pipes and solved with FFMPEG_FLAGS=-fflags +flush_packets+genpts+discardcorrupt+nobuffer .

Anyways, I would like to please to see other user experience to tune FFMPEG flags (e.g. -flags low_delay, ...etc.).....as I see in FFMPEG website MANY  parameters....even it can run as RTSP server!!

How to tune, how to re-order frames, how all cams be on sync when play the recorded files,....etc.

Thanks in advance for all ideas & experiences....

**Comment by mrlt8** [comment]
Thanks @AdiAbuAli! I added the flags back to the bridge in v1.0.1: https://github.com/mrlt8/docker-wyze-bridge/blob/d562b41afa03d858061986662fdbf2ae3017e4cb/app/wyze_bridge.py#L392

While the bridge will use the flags by default, you can still override them with `FFMPEG_FLAGS` as usual.

**Reply by AdiAbuAli** [reply]
Thanks Mrlt8...

**Comment by AdiAbuAli** [comment]
I tried libx264 -preset ultrafast parameters in FFMPEG command, and the result was fabulous.....anyways, I stopped it later, as CPU jumped 4x !

**Comment by AdiAbuAli** [comment]
To whom it may interest...
I noticed the stream is much smoother if you lower Log level to -loglevel error...specialy if you are running on low PC specs, and/or want to overcome Timestamp not set in stream 0 warning message by setting (-use_wallclock_as_timestamps 1)....
seems this will reduce overhead process on FFMPEG.....

Just sharing experience....

**Reply by mrlt8** [reply]
Log level is already the default behavior:

https://github.com/mrlt8/docker-wyze-bridge/blob/a5e88942e8b27d282a7483c0842433390683f84b/app/wyze_bridge.py#L405
